### DISCUSSION TITLE
Customizing editor (tab's) label/title text, and/or its format?

### DISCUSSION THREAD

**Discussion by peter-lyons-kehl** [microsoft/vscode-discussions#60] [Extension Development QnA] [answered]
In short: Could a VS Code extension either
 - add a custom format to options in `workbench.editor.labelFormat`, and/or
 - modify format (the text, and/or bold/large formatting) of the source code editor tab titles/labels?

(If it makes it any easier, it's needed for Rust files only - with `.rs` extension, or - even better - only to files called `mod.rs`. Otherwise applying this to any source file tabs is OK, too.)

If so, where under https://code.visualstudio.com/api, or what parts of the API's/what approach would be the smoothest and simplest, please?

(Insiders/Unstable/GIT-based-only API is fine, if it enables this.)

(I'm completely new to VS Code extensions API.
Longer: Rust mainstream has all (sub)modules in files called `mod.rs`. The module name/path is based on the directory path. Switching between even a few modules in VS Code is then a pain ( regardless of `workbench.editor.labelFormat`). I've searched at Market Place, and at its "tab" and "tabs" tag, but nothing relevant.
I believe this would make Rust community grateful.)

Update: Thanks to https://stackoverflow.com/questions/62569941/show-only-parent-folder-names, I'll experiment with [Monkey Patch](https://marketplace.visualstudio.com/items?itemName=iocave.monkey-patch) and will try to resurrect https://github.com/skoshy/nice-index-vscode. (There is also https://github.com/microsoft/vscode/issues/41909, but seems ignored.)

**Comment by isidorn** [comment, marked as answer]
This is currently not possible. And this request seems to be covered with https://github.com/microsoft/vscode/issues/41909 as you already found out.

**Comment by matthew-dean** [comment]
The current answer is not correct. You can do this! See: https://code.visualstudio.com/docs/getstarted/userinterface#_customize-tab-labels
